welly/well.py

```diff
@@ -289,53 +229,168 @@ def from_las(cls,
+        index_curves = ['DEPT', 'TIME']
+
+        # unpack datasets
+        for name, (df_data, df_header) in datasets.items():
+
+            # The default behaviour is to keep welly's current behaviour, which
+            # is to:
+            # (1) assume the first column from data is the depth/time curve AND
+            # (2) assume that lasio was able to recognise the depth unit
+
+            # retrieve well parameters from header
+            well_params = {}
+            for field, (sect, code) in LAS_FIELDS['data'].items():
+                well_params[field] = utils.lasio_get(df_header,
+                                                     sect, code,
+                                                     remap=remap,
+                                                     funcs=funcs)
+
+            # get index unit
             try:
-                data = urllib.request.urlopen(fname).read().decode()
-            except urllib.error.HTTPError as e:
-                raise WellError('Could not retrieve url: ', e)
-            fname = (StringIO(data))
-
-        las = lasio.read(fname, encoding=encoding)
-
-        # Pass to other constructor.
-        return cls.from_lasio(las,
-                              remap=remap,
-                              funcs=funcs,
-                              data=data,
-                              req=req,
-                              alias=alias,
-                              fname=fname,
-                              index=index)
+                index_unit = df_header[(df_header["section"] == name)].iloc[0].unit
+            except AttributeError:
+                index_unit = ''
+
+            # get depth/time array
+            l_index = df_data.iloc[:, 0].values
+
+            if index:
+                if (index_unit == 'm' or index_unit == '') and "ft" in index.lower():
+                    l_index = l_index * 3.280839895  # convert to ft
+                    index_unit = 'ft'
+                elif (index_unit == 'ft' or index_unit == '') and "m" in index.lower():
+                    l_index = l_index * 0.3048000000012192  # convert to m
+                    index_unit = 'm'
+                else:
+                    raise KeyError("Index must be 'm', or 'ft' but {} was "
+                                   "given.".format(index_unit))
+
+            if l_index[0] < l_index[1]:
+                well_params['depth'] = l_index
+            else:
+                well_params['depth'] = np.flipud(l_index)
+
+            well_params['basis_units'] = index_unit
+
+            # retrieve curve mnemonic and units from header
+            curve_units = df_header[
+                (df_header["section"] == name)][['mnemonic', 'unit', 'descr']].set_index('mnemonic').T
+
+            if req and alias:
+                req = utils.flatten_list([v for k, v in alias.items() if k in req])
+
+            if data and req:
+                curves = {m: Curve.from_lasio_curve(
```

Comment:
"m" is not very descriptive, I would stay with previous mnemonic, or something more descriptive than "m"